conversion/indico_conversion/conversion.py

```diff
@@ -98,20 +102,80 @@ def _process(self):
         elif request.form['status'] != '1':
             ConversionPlugin.logger.error('Received invalid status %s for %s', request.form['status'], attachment)
             return jsonify(success=False)
-        name, ext = os.path.splitext(attachment.file.filename)
-        title = get_pdf_title(attachment)
-        pdf_attachment = Attachment(folder=attachment.folder, user=attachment.user, title=title,
-                                    description=attachment.description, type=AttachmentType.file,
-                                    protection_mode=attachment.protection_mode, acl=attachment.acl)
-        data = request.files['content'].stream.read()
-        pdf_attachment.file = AttachmentFile(user=attachment.file.user, filename=f'{name}.pdf',
-                                             content_type='application/pdf')
-        pdf_attachment.file.save(data)
-        db.session.add(pdf_attachment)
-        db.session.flush()
-        pdf_state_cache.set(str(attachment.id), 'finished', timeout=timedelta(minutes=15))
-        ConversionPlugin.logger.info('Added PDF attachment %s for %s', pdf_attachment, attachment)
-        signals.attachments.attachment_created.send(pdf_attachment, user=None)
+        pdf = request.files['content'].stream.read()
+        save_pdf(attachment, pdf)
+        return jsonify(success=True)
+
+
+@celery.task(bind=True, max_retries=3)
+def submit_attachment_cloudconvert(task, attachment):
+    """Sends an attachment's file to the CloudConvert conversion service"""
+    from indico_conversion.plugin import ConversionPlugin
+    if ConversionPlugin.settings.get('maintenance'):
+        task.retry(countdown=900)
+
+    cloudconvert.configure(api_key=ConversionPlugin.settings.get('cloudconvert_api_key'), sandbox=True)
```

Comment:
using global state for this is ew. `configure` calls this:

```python
def set_config(options=None, **config):
    """Create new default api object with given configuration
    """
    global __client__
    __client__ = CloudConvertRestClient(options or {}, **config)
    return __client__
```

I'm not sure why they are doing such obscure things. I opened an issue cloudconvert/cloudconvert-python#26 about this...